### PR TITLE
[FLINK-16961] Bump Netty 4.X to 4.1.44

### PR DESCRIPTION
**Pull request by zentol**
To address multiple CVEs in netty.

### CODE REVIEW

**Comment by zentol** [comment]
Any input on whether to backport this PR to 1.10/1.9 would be appreciated.

File: flink-connectors/flink-connector-hive/pom.xml [hunk 1]
```diff
@@ -934,6 +934,42 @@ under the License.
...
+							<!-- Override arrow netty dependency -->
+							<groupId>io.netty</groupId>
+							<artifactId>netty-buffer</artifactId>
+						</exclusion>
+						<exclusion>
+							<!-- Override arrow netty dependency -->
+							<groupId>io.netty</groupId>
+							<artifactId>netty-common</artifactId>
+						</exclusion>
+					</exclusions>
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-connector-hive/pom.xml:954]: the dependency convergence complains because we did not explicitly handle the mismatch.

File: flink-connectors/flink-connector-hive/pom.xml [hunk 2]
```diff
@@ -934,6 +934,42 @@ under the License.
...
+				</dependency>
+
+				<dependency>
+					<!-- Bump arrow netty dependency -->
+					<groupId>io.netty</groupId>
+					<artifactId>netty-common</artifactId>
+					<version>4.1.44.Final</version>
+					<scope>provided</scope>
+				</dependency>
+			</dependencies>
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-connector-hive/pom.xml:971]: did you activate the `hive-3.1.1` profile?

**Comment by zentol** [comment]
> I was wondering whether we shouldn't declare minimum requirements in the parent pom's dependency management section

DependencyManagement has subtle flaws that make it undesirable for this case. This will likely also apply to other cases and existing depMgmt entries that we currently have, which we may want to re-evaluate at some point.

The core issue is that dependency management entries do not affect the published poms.

Let's say you have a module M, depending on A, depending on B 1.0, and M has a depMgmt entry for B setting it to 1.1.
While you are working in the module, writing code, running tests, packaging dependencies, B is set to 1.1 as expected.
But, in the published pom there is still just the dependency on A. You don't see anything about B, so a user would pull in B 1.0 again DependencyManagement is not transitive, i.e., not visible to downstream modules. See also MNG-5761.
This is also why we have so many entries in the root pom; if one module has a depMgmt entry, and another module depends on it, then the second one doesn't see the depMgmt entries of the first one, and has to handle the conflicts locally again.

In other words, if you bundle a dependency affected by depMgmt entries then you're good. But if this dependency is exposed to users, then you have achieved no practical benefit.
A security scanner running over the project itself will be happy, a scanner running over the published artifacts will complain.

File: flink-python/src/main/resources/META-INF/NOTICE [hunk 3]
```diff
@@ -20,13 +20,13 @@ This project bundles the following dependencies under the Apache Software Licens
...
 - io.netty:netty-buffer:4.1.34.Final
 - io.netty:netty-codec:4.1.34.Final
 - io.netty:netty-codec-http:4.1.34.Final
 - io.netty:netty-codec-http2:4.1.34.Final
 - io.netty:netty-codec-socks:4.1.34.Final
-- io.netty:netty-common:4.1.27.Final
+- io.netty:netty-common:4.1.44.Final
 - io.netty:netty-common:4.1.34.Final
 - io.netty:netty-handler:4.1.34.Final
 - io.netty:netty-handler-proxy:4.1.34.Final
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-python/src/main/resources/META-INF/NOTICE:32]: These are bundled in `beam-vendor-grpc-1_21_0`, and I have no idea how/whether we can update these.

[the remark this answers is not in the document]
Comment (zentol) [reply flink-python/src/main/resources/META-INF/NOTICE:32]: I did ignore them tho and had to do some research first, good job pointing them out. 👍

File: flink-connectors/flink-connector-cassandra/src/main/resources/META-INF/NOTICE [hunk 4]
```diff
@@ -9,8 +9,8 @@ This project bundles the following dependencies under the Apache Software Licens
...
-- io.netty:netty-handler:4.0.33.Final
-- io.netty:netty-buffer:4.0.33.Final
-- io.netty:netty-common:4.0.33.Final
-- io.netty:netty-transport:4.0.33.Final
-- io.netty:netty-codec:4.0.33.Final
+- io.netty:netty-handler:4.1.44.Final
+- io.netty:netty-buffer:4.1.44.Final
+- io.netty:netty-common:4.1.44.Final
+- io.netty:netty-transport:4.1.44.Final
+- io.netty:netty-codec:4.1.44.Final
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-connector-cassandra/src/main/resources/META-INF/NOTICE:16]: I don't see `netty-resolver` in the dependency or shade-plugin output.

File: flink-connectors/flink-connector-elasticsearch5/pom.xml [hunk 5]
```diff
@@ -82,6 +82,55 @@ under the License.
 			<version>${elasticsearch.version}</version>
 		</dependency>
 
+		<dependency>
+			<!-- Bump elasticsearch netty dependency -->
+			<groupId>io.netty</groupId>
+			<artifactId>netty-buffer</artifactId>
+			<version>4.1.44.Final</version>
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-connector-elasticsearch5/pom.xml:89]: Smallest required jump.

[the remark this answers is not in the document]
Comment (tillrohrmann) [reply flink-connectors/flink-connector-elasticsearch5/pom.xml:89]: Hmm ok. I was just thinking if already do the effort to upgrade why not trying the latest version. But on the other hand, it might cause more problems.

File: flink-connectors/flink-connector-hive/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tillrohrmann) [reply flink-connectors/flink-connector-hive/pom.xml:971]: No I did not. If this explains then I guess it is correct.

File: flink-connectors/flink-connector-cassandra/src/main/resources/META-INF/NOTICE (hunk 4 above)
[the remark this answers is not in the document]
Comment (tillrohrmann) [reply flink-connectors/flink-connector-cassandra/src/main/resources/META-INF/NOTICE:16]: `netty-transport ` has a compile time dependency. Hence it should be included in the final artifact: https://mvnrepository.com/artifact/io.netty/netty-transport/4.1.44.Final

File: flink-connectors/flink-connector-hive/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tillrohrmann) [reply flink-connectors/flink-connector-hive/pom.xml:954]: I see. Thanks for the info.

**Comment by tillrohrmann** [comment]
Thanks for answering my questions @zentol.

I think the last thing to do is to address the modules which have also a netty dependency and resolving the `netty-resolver` dependency.

File: flink-connectors/flink-connector-hive/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-connector-hive/pom.xml:971]: This dependencies block is in the `hive-3.1.1` profile, so it is kinda important ;)

File: flink-connectors/flink-connector-cassandra/src/main/resources/META-INF/NOTICE (hunk 4 above)
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-connector-cassandra/src/main/resources/META-INF/NOTICE:16]: ah, i see it now. I must've checked on the wrong branch.

File: flink-connectors/flink-connector-hive/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tillrohrmann) [reply flink-connectors/flink-connector-hive/pom.xml:971]: Seems to be correct given `mvn dependency:tree -Phive-3.1.1`.

**Comment by zentol** [comment]
@tillrohrmann I believe the remaining issues are solved. Could you double-check that netty no longer shows up in the sql connectors? I couldn't reproduce this yet myself :/
